File: C/Cap03/c03ex04.c
// c03ex04.c

#include <stdio.h>

int main(void) {
  double HT, VH, PD, SB, TD, SL;

  printf("Entre quantidade de horas trabalhadas ..: ");
  scanf("%lf", &HT);
  while (getchar() != '\n');

  printf("Entre valor do salario-hora ............: ");
  scanf("%lf", &VH);
  while (getchar() != '\n');

  printf("Entre valor do percentual de desconto ..: ");
  scanf("%lf", &PD);
  while (getchar() != '\n');

  SB = HT * VH;
  TD = (PD / 100) * SB;
  SL = SB - TD;

  printf("\n");
  printf("Salario Bruto ....: %9.2lf\n", SB);
  printf("Salario Liquido ..: %9.2lf\n", SL);

  printf("\nTecle <Enter> para encerrar... ");
  while (getchar() != '\n');

  return 0;
}

File: DART/Cap03/c03ex04.dart
// c03ex04.dart

import 'dart:io';

String fmtNumero(double numero, int totalDigitos, 
                 int decimalDigitos) {
  final int inteiro = totalDigitos - decimalDigitos - 1;
  final int decimal = decimalDigitos;
  final String nrFormatado = numero.toStringAsFixed(decimal).
    replaceAll('.', ',');
  final List<String> parte = nrFormatado.split(',');
  final String inteiroParte = parte[0].padLeft(inteiro, ' ');
  final String decimalParte = parte[1].padRight(decimal, '0');
  return '$inteiroParte.$decimalParte';
}

void main() {
  double ht;
  double vh;
  double pd;
  double sb;
  double td;
  double sl;

  stdout.write("Entre quantidade de horas trabalhadas ..: ");
  ht = double.tryParse(stdin.readLineSync() ?? '') ?? 0.0;

  stdout.write("Entre valor do salario-hora ............: ");
  vh = double.tryParse(stdin.readLineSync() ?? '') ?? 0.0;

  stdout.write("Entre valor do percentual de desconto ..: ");
  pd = double.tryParse(stdin.readLineSync() ?? '') ?? 0.0;

  sb = ht * vh;
  td = (pd / 100) * sb;
  sl = sb - td;

  print("\nSalario Bruto ....: ${fmtNumero(sb, 9, 2)}");
  print("Salario Liquido ..: ${fmtNumero(sl, 9, 2)}");

  stdout.write("\nTecle <Enter> para encerrar... ");
  stdin.readLineSync();
}
